src/scripts/circleci-job-canceller/long_job_canceller.py

```diff
@@ -8,73 +9,97 @@
 
-              if ( (created_at < (now - job_midlife_warning)) and (created_at > (now - job_life_clock))):
-                  username = get_workflow_started_by(current_workflow)
-                  if username in robot_committers:
-                      continue
+        for current_workflow in get_all_items(f"/pipeline/{current_pipeline['id']}/workflow", headers, None):
+            if current_workflow["status"] == "on_hold":
+                created_at_str = current_workflow["created_at"]
+                created_at = isoparse(created_at_str)
 
-                  yield { "job_status": "age_warning", "name": current_workflow['name'], "id": current_workflow['id'], "username": username }
+                if ((created_at < (now - job_midlife_warning)) and (created_at > (now - job_life_clock))):
+                    username = get_workflow_started_by(
+                        current_workflow, headers)
+                    if username in robot_committers:
+                        continue
 
-              if (created_at < (now - job_life_clock)):
-                  username = get_workflow_started_by(current_workflow)
-                  yield { "job_status": "too_old", "name": current_workflow['name'], "id": current_workflow['id'], "username": username }
-
-
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    parser.add_argument("circleapitoken", help="the CircleCI API token for this script")
-    parser.add_argument("orgreposlug", help="the location, user-or-org/repository-name , of this repository")
+                    yield {"job_status": "age_warning", "name": current_workflow['name'], "id": current_workflow['id'], "username": username}
 
-    parser.add_argument("--output-file", help="output to file path", default="/tmp/notifications.tsv", )
-    parser.add_argument("--just-do-it", help="just cancel jobs", default=False, action="store_true")
+                if (created_at < (now - job_life_clock)):
+                    username = get_workflow_started_by(
+                        current_workflow, headers)
+                    yield {"job_status": "too_old", "name": current_workflow['name'], "id": current_workflow['id'], "username": username}
 
-    args = parser.parse_args()
 
-    CIRCLE_API_KEY = args.circleapitoken
-    cancel_jobs_here = args.just_do_it
+def main(circleapitoken, orgreposlug, output_file, commit):
+    standard_headers = {"Circle-Token": circleapitoken}
 
-    standard_headers = {"Circle-Token": args.circleapitoken}
+    simple_path = os.path.abspath(os.path.expanduser(
+        os.path.expandvars(output_file)))
 
-    simple_path = os.path.abspath(os.path.expanduser(os.path.expandvars(args.output_file)))
     with open(simple_path, 'w') as f:
         f.write("job_status\tproceed\tid\tusername\tname\n")
-        for current_info in find_old_workflow_ids(args.orgreposlug):
+        for current_info in find_old_workflow_ids(
+            orgreposlug,
+            now - (job_life_clock * 5),
```

Comment:
wait what why is it times 5?

Reply:
x5 purely for a window to look back across 🤷 magic number for sure